src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -207,9 +230,16 @@ private TrieNode searchForJsonPathKeyNode(byte[] bytes, Iterator<MaskingState.Se
             if (node == null) {
                 return null;
             }
-            MaskingState.SegmentReference segmentReference = jsonPath.next();
+            SegmentReference segmentReference = jsonPath.next();
             int keyStartIndex = segmentReference.start;
             int keyLength = segmentReference.offset;
+            if (segmentReference.isArraySegment()) {
+                node = node.children[Character.forDigit(keyStartIndex, 10) + BYTE_OFFSET];
```

Comment:
`10` is a magic number. Please make it a named constant

Reply:
Ok.